flocker/node/agents/blockdevice.py

```diff
@@ -53,6 +59,125 @@ class UnattachedVolume(VolumeException):
     """
 
 
+DATASET = Field(
+    u"dataset",
+    lambda dataset: dataset.dataset_id,
+    u"The unique identifier of a dataset."
+)
+
+DATASET_ID = Field(
+    u"dataset_id",
+    lambda dataset_id: unicode(dataset_id),
```

Comment:
We made it a `UUID` instance on the new backend type (and then maybe re-used this field for both kinds of things ...)